talpid-core/src/firewall/macos.rs

```diff
@@ -566,6 +582,45 @@ impl Firewall {
         Ok(rules)
     }
 
+    /// Generate rules that allow traffic to the networks required for Apple push notification
+    /// services to work. This is a hack to get around the fact that apple services in MacOS 15 has
+    /// a bug where they don't respect the routing table.
+    ///
+    /// All allowed networks are part of apple-owned IP subnets.
+    fn get_apple_services_bypass_rules(&self) -> Result<Vec<pfctl::FilterRule>> {
+        // https://support.apple.com/en-us/102266
+        let apple_networks: &[IpNetwork] = &[
+            "17.249.0.0/16".parse().unwrap(),
+            "17.252.0.0/16".parse().unwrap(),
+            "17.57.144.0/22".parse().unwrap(),
+            "17.188.128.0/18".parse().unwrap(),
+            "17.188.20.0/23".parse().unwrap(),
+            "2620:149:a44::/48".parse().unwrap(),
+            "2403:300:a42::/48".parse().unwrap(),
+            "2403:300:a51::/48".parse().unwrap(),
+            "2a01:b740:a42::/48".parse().unwrap(),
+        ];
+
+        let apple_ports: &[u16] = &[443, 2197, 5223];
+
+        let mut rules = vec![];
+        for &net in apple_networks {
+            for &port in apple_ports {
+                let mut rule_builder = self.create_rule_builder(FilterRuleAction::Pass);
+                rule_builder.quick(true);
+                let allow_out = rule_builder
```

Comment:
you should add a further constraint for `user = 0` on this rule so that only root processes (such as `apsd`) can send off-vpn traffic to these ips

Reply:
Good idea, but were currently working on getting rid of this feature all together.

Reply:
Cool! What's your new approach?

Reply:
Interesting! what's your new approach for unblocking apple services?